projects/RabbitMQ.Client/client/impl/SocketFrameHandler.cs

```diff
@@ -186,8 +199,8 @@ public void Close()
                 {
                     try
                     {
-                        _frameChannel.Writer.Complete();
-                        _frameWriter.Wait();
+                        _channelWriter.Complete();
+                        _writerTask.Wait();
```

Comment:
I would prefer here

```
_writerTask.GetAwaiter().GetResult()
```

I know for this code path it doesn't make a big difference because we try catch anyway all the exceptions but then at least we slowly move to a standardized way of do sync over async